Source/Core/Core/PowerPC/SignatureDB/MEGASignatureDB.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "Common/CommonTypes.h"
+#include "Core/PowerPC/PPCSymbolDB.h"
+
+struct MEGASignatureReference
+{
+  u32 offset;
+  std::string name;
+
+  MEGASignatureReference(u32 _offset, std::string _name) : offset(_offset), name(_name) {}
+};
+
+struct MEGASignature
+{
+  std::vector<u32> code;
+  std::string name;
+  std::vector<MEGASignatureReference> refs;
+};
+
+class MEGASignatureDB
```

Comment:
This header should probably have documentation on the format itself. This should also be using the general inheritance format that the other two signature DB handlers are using (DSY and CSV), otherwise this doesn't follow the API contract. If the current API doesn't map to mega files that well, then the API should be changed instead of simply not using it.